In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
mep = pd.read_csv('retweets.csv')

In [3]:
mep.shape

(26133, 21)

In [4]:
mep.columns

Index(['origUserId', 'origUserScreenName', 'origMepId', 'origMepName',
       'origMepGroupId', 'origMepGroupShort', 'origMepCountryId',
       'origMepCountryShort', 'retweetUserId', 'retweetUserScreenName',
       'retweetMepId', 'retweetMepName', 'retweetMepGroupId',
       'retweetMepGroupShort', 'retweetMepCountryId', 'retweetMepCountryShort',
       'origCreatedAt', 'origTweetId', 'retweetCreatedAt', 'retweetTweetId',
       'lang'],
      dtype='object')

Delete the records with not English languages

In [5]:
mep.lang

0         el
1         el
2         fr
3         en
4         fr
        ... 
26128     en
26129    und
26130     fr
26131     fr
26132     es
Name: lang, Length: 26133, dtype: object

In [6]:
mep = mep.loc[mep.lang == 'en']

In [7]:
import tweepy
auth = tweepy.OAuthHandler('hsaHW0P7kFtP7yiOTWV6E5Cmz', 'A1fgwDsjBQ48dYM8xHqq1fafs8sJpOAj9m9QFkEmqBjorpEk4X')

In [8]:
mep

origUserId origUserScreenName  origMepId    origMepName  \
3        19017675       Nigel_Farage       4525   Nigel FARAGE   
5        17675072       MartinSchulz       1911  Martin SCHULZ   
11       17675072       MartinSchulz       1911  Martin SCHULZ   
12      744743599           Tim_Aker      99650       Tim AKER   
13      744743599           Tim_Aker      99650       Tim AKER   
...           ...                ...        ...            ...   
26121   202610289           akhanmep     124962     Afzal KHAN   
26122   202610289           akhanmep     124962     Afzal KHAN   
26126   202610289           akhanmep     124962     Afzal KHAN   
26127   202610289           akhanmep     124962     Afzal KHAN   
26128    85794542       DanHannanMEP       4555  Daniel HANNAN   

       origMepGroupId origMepGroupShort  origMepCountryId origMepCountryShort  \
3                   6              EFDD                27                 GBR   
5                   1               S&D                10                 DEU   
11                  1               S&D                10                 DEU   
12                  6              EFDD                27                 GBR   
13                  6              EFDD                27                 GBR   
...               ...               ...               ...                 ...   
26121               1               S&D                27                 GBR   
26122               1               S&D                27                 GBR   
26126               1               S&D                27                 GBR   
26127               1               S&D                27                 GBR   
26128               5               ECR                27                 GBR   

       retweetUserId retweetUserScreenName  ...   retweetMepName  \
3          121171051        MargotLJParker  ...    Margot PARKER   
5         1668992125         NathanGillMEP  ...      Nathan GILL   
11          21648649         maritaulvskog  ...   Marita ULVSKOG   
12         121171051        MargotLJParker  ...    Margot PARKER   
13          19017675          Nigel_Farage  ...     Nigel FARAGE   
...              ...                   ...  ...              ...   
26121      829060856            TheresaMEP  ...  Theresa GRIFFIN   
26122     1354570123              julie4nw  ...       Julie WARD   
26126      829060856            TheresaMEP  ...  Theresa GRIFFIN   
26127     1354570123              julie4nw  ...       Julie WARD   
26128      121171051        MargotLJParker  ...    Margot PARKER   

      retweetMepGroupId  retweetMepGroupShort retweetMepCountryId  \
3                     6                  EFDD                  27   
5                     6                  EFDD                  27   
11                    1                   S&D                  26   
12                    6                  EFDD                  27   
13                    6                  EFDD                  27   
...                 ...                   ...                 ...   
26121                 1                   S&D                  27   
26122                 1                   S&D                  27   
26126                 1                   S&D                  27   
26127                 1                   S&D                  27   
26128                 6                  EFDD                  27   

       retweetMepCountryShort                   origCreatedAt  \
3                         GBR  Wed Jul 04 11:05:03 +0000 2012   
5                         GBR  Fri Jul 05 08:00:23 +0000 2013   
11                        SWE  Sat May 17 12:47:17 +0000 2014   
12                        GBR  Wed May 28 16:47:44 +0000 2014   
13                        GBR  Wed May 28 16:47:44 +0000 2014   
...                       ...                             ...   
26121                     GBR  Sun Jan 24 14:39:12 +0000 2016   
26122                     GBR  Sun Jan 24 14:39:12 +0000 2016   
26126     

In [9]:
origTweetList = mep.origTweetId.tolist()

In [10]:
try:
    redirect_url = auth.get_authorization_url()
except tweepy.TweepError:
    print('Error! Failed to get request token.')

In [11]:
# api = tweepy.API(auth)
# origText = []
# for tweetid in origTweetList:
#     try:
#         tweet = api.get_status(tweetid)
#         origText.append(tweet.text)
#         print(tweet.text)
#     except:
#         origText.append('')
#         print(' ')

https://stackoverflow.com/questions/44581647/retrieving-a-list-of-tweets-using-tweet-id-in-tweepy

In [12]:
import tweepy

import time

t0 = time.time()

def lookup_tweets(tweet_IDs, api):
    full_tweets = []
    tweet_count = len(tweet_IDs)
    try:
        for i in range((tweet_count // 100) + 1):
            # Catch the last group if it is less than 100 tweets
            end_loc = min((i + 1) * 100, tweet_count)
            full_tweets.extend(
                api.statuses_lookup(id_=tweet_IDs[i * 100:end_loc])
            )  
        return full_tweets
    except tweepy.TweepError:
        print('Something went wrong, quitting...')

auth = tweepy.OAuthHandler('hsaHW0P7kFtP7yiOTWV6E5Cmz', 'A1fgwDsjBQ48dYM8xHqq1fafs8sJpOAj9m9QFkEmqBjorpEk4X')

api = tweepy.API(auth, wait_on_rate_limit=True, wait_on_rate_limit_notify=True)

# do whatever it is to get por.TweetID - the list of all IDs to look up

results = lookup_tweets(origTweetList, api)

final_dic = {}

t1 = time.time()
total = t1-t0
print(total)

280.1609683036804


In [25]:
for tweet in results:
    if tweet:
        final_dic[tweet.id] = tweet.text

In [26]:
import pprint
# pprint.pprint(final_dic)
len(final_dic)

10126

In [27]:
tweet_text = pd.DataFrame.from_dict(final_dic, orient='index', columns=['tweetText'])

## Cross Validate that the number of rows is **10126**

### Something to investigate: Do we have duplicate tweets on the dataset?
### mep rows: 13796
### mep.origTweetId: 10834

In [29]:
tweet_text

tweetText
519050739642286080  Lord Hill's written answers to EP are now on m...
519139901359783936  Our @SebDance asks @ABratusek about global cli...
518024241024073728  I write for @LabourList Tory threat to Court o...
519109106272317440  “@Ashleyfoxmep: and named her party "the Allia...
518721908951769088  Good luck to @TheProgressives in Bulgaria toda...
...                                                               ...
690861330555887616  Memo to Tories for Reform in EU: Never mind th...
690514426991120384  #EUreferendum is going to be about working peo...
690096989292367872  Extremely rich corporate folk who do extremely...
690207644838133760  I have called upon the Indian government to re...
690928472223563777  Proud to sign @Grassroots_Out's Kettering Decl...

[10126 rows x 1 columns]

In [68]:
mepExt = pd.merge(mep, tweet_text, left_on='origTweetId', right_index=True , how='inner')

In [84]:
group = mepExt.groupby('origMepGroupShort').count()
groupFiltered = group.loc[group.origUserId > 50]
mepExt = mepExt.loc[mepExt.origMepGroupShort.isin(list(groupFiltered.index))]

In [86]:
mepExt

origUserId origUserScreenName  origMepId           origMepName  \
3        19017675       Nigel_Farage       4525          Nigel FARAGE   
5        17675072       MartinSchulz       1911         Martin SCHULZ   
11       17675072       MartinSchulz       1911         Martin SCHULZ   
12      744743599           Tim_Aker      99650              Tim AKER   
13      744743599           Tim_Aker      99650              Tim AKER   
...           ...                ...        ...                   ...   
26114  1485429175     JSaryuszWolski      28297  Jacek SARYUSZ-WOLSKI   
26117   829060856         TheresaMEP     124961       Theresa GRIFFIN   
26118   829060856         TheresaMEP     124961       Theresa GRIFFIN   
26119   829060856         TheresaMEP     124961       Theresa GRIFFIN   
26128    85794542       DanHannanMEP       4555         Daniel HANNAN   

       origMepGroupId origMepGroupShort  origMepCountryId origMepCountryShort  \
3                   6              EFDD                27                 GBR   
5                   1               S&D                10                 DEU   
11                  1               S&D                10                 DEU   
12                  6              EFDD                27                 GBR   
13                  6              EFDD                27                 GBR   
...               ...               ...               ...                 ...   
26114               4               EPP                20                 POL   
26117               1               S&D                27                 GBR   
26118               1               S&D                27                 GBR   
26119               1               S&D                27                 GBR   
26128               5               ECR                27                 GBR   

       retweetUserId retweetUserScreenName  ...  retweetMepGroupId  \
3          121171051        MargotLJParker  ...                  6   
5         1668992125         NathanGillMEP  ...                  6   
11          21648649         maritaulvskog  ...                  1   
12         121171051        MargotLJParker  ...                  6   
13          19017675          Nigel_Farage  ...                  6   
...              ...                   ...  ...                ...   
26114      539156512       ZdzKrasnodebski  ...                  5   
26117     1354570123              julie4nw  ...                  1   
26118      202610289              akhanmep  ...                  1   
26119     1354570123              julie4nw  ...                  1   
26128      121171051        MargotLJParker  ...                  6   

      retweetMepGroupShort  retweetMepCountryId retweetMepCountryShort  \
3                     EFDD                   27                    GBR   
5                     EFDD                   27                    GBR   
11                     S&D                   26                    SWE   
12                    EFDD                   27                    GBR   
13                    EFDD                   27                    GBR   
...                    ...                  ...                    ...   
26114                  ECR                   20                    POL   
26117                  S&D                   27                    GBR   
26118                  S&D                   27                    GBR   
26119                  S&D                   27                    GBR   
26128                 EFDD                   27                    GBR   

                        origCreatedAt         origTweetId  \
3      Wed Jul 04 11:05:03 +0000 2012  220473289259233285   
5      Fri Jul 05 08:00:23 +0000 2013  353060776707235841   
11     Sat May 17 12:47:17 +0000 2014  467647542378123264   
12     Wed May 28 16:47:44 +0000 2014  471694322472353793   
13     Wed May 28 16:47:44 +0000 2014  471694322472353793   
...                               ...                 ...   
